backend/dal/Repositories/DocumentRepository.cs

```diff
@@ -80,6 +80,11 @@ public bool Delete(PimsDocument document)
         {
             document.ThrowIfNull(nameof(document));
 
+            foreach (var pimsActivityInstanceDocument in document.PimsActivityInstanceDocuments)
```

Comment:
changed my mind on this, if we are deleting a document we should delete any dependent relationships. This is a stand-in for proper cascades.